rust/src/lib/ovn.rs

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Deserializer, Serialize};
+use std::collections::HashMap;
+use std::fmt;
+use std::str::FromStr;
+
+pub const OVN_BRIDGE_MAPPINGS: &str = "ovn-bridge-mappings";
+
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize)]
+#[non_exhaustive]
+pub struct OvnConfiguration {
+    #[serde(
+        rename = "bridge-mappings",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub bridge_mappings: Option<Vec<OvnBridgeMapping>>,
+    #[serde(skip)]
+    pub(crate) prop_list: Vec<&'static str>,
+}
+
+impl OvnConfiguration {
+    pub fn is_none(&self) -> bool {
+        self.bridge_mappings.is_none()
+    }
+}
+
+impl<'de> Deserialize<'de> for OvnConfiguration {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let mut ovn_conf = Self::default();
+        let v = serde_json::Value::deserialize(deserializer)?;
+
+        if let Some(v) = v.as_object() {
+            if let Some(v) = v.get("bridge-mappings") {
+                ovn_conf.prop_list.push("bridge-mappings");
+                match value_to_ovn_bridge_mapping(v) {
+                    Ok(bridge_mappings) => {
+                        ovn_conf.bridge_mappings = bridge_mappings
+                    }
+                    Err(err) => {
+                        return Err(serde::de::Error::custom(format!(
+                        "Error unmarshalling `OvnBridgeMapping` {v:?}: {err}"
```

Comment:
Normally we do not scare people with debug output `{:?}`.

Reply:
Hm, I had just copied this from https://github.com/nmstate/nmstate/blob/f0f7c1edd8152dc03e6287c2222b524167c6f128/rust/src/lib/ovs.rs#L114

Well, doesn't matter, I'm trying to see if I can avoid using the custom deserializer - if so, all this code will be gone.

Reply:
This was removed, since the deserializer code is now gone.